test/gtest/uct/test_error_handling.cc

```diff
@@ -0,0 +1,136 @@
+
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2001-2016.  ALL RIGHTS RESERVED.
+* Copyright (C) UT-Battelle, LLC. 2016. ALL RIGHTS RESERVED.
+* Copyright (C) ARM Ltd. 2016.All rights reserved.
+* See file LICENSE for terms.
+*/
+
+extern "C" {
+#include <uct/api/uct.h>
+}
+#include <common/test.h>
+#include "uct_test.h"
+
+class test_error_handling : public uct_test {
+public:
+    virtual void init() {
+        uct_test::init();
+
+        m_e1 = uct_test::create_entity(0);
+        m_entities.push_back(m_e1);
+
+        m_e2 = uct_test::create_entity(0);
+        m_entities.push_back(m_e2);
+
+        set_config("LOG_LEVEL=fatal");
+        connect();
+    }
+
+    static ucs_status_t am_dummy_handler(void *arg, void *data, size_t length, void *desc) {
+        return UCS_OK;
+    }
+
+    static ucs_status_t pending_cb(uct_pending_req_t *self)
+    {
+        req_count++;
+        return UCS_OK;
+    }
+
+    void connect() {
+        m_e1->connect(0, *m_e2, 0);
+        m_e2->connect(0, *m_e1, 0);
+
+        uct_iface_set_am_handler(m_e1->iface(), 0, am_dummy_handler,
+                                 NULL, UCT_AM_CB_FLAG_SYNC);
+        uct_iface_set_am_handler(m_e2->iface(), 0, am_dummy_handler,
+                                 NULL, UCT_AM_CB_FLAG_SYNC);
+    }
+
+protected:
+    static int req_count;
+    entity *m_e1, *m_e2;
+};
+
+int test_error_handling::req_count = 0;
+
+UCS_TEST_P(test_error_handling, peer_failure)
```

Comment:
Yes, it'd be good. But this test is intended to be executed for all transports supporting error handling (UD, UGNI?). 
Can we use GetParam() to check what transport is being used in run-time? Then we could add transport specific, right?